src/main/scala/cromwell/engine/WorkflowManager.scala

```diff
@@ -84,10 +88,16 @@ trait WorkflowManagerActor extends Actor with WorkflowManager {
     case WorkflowStatus(id) =>
       val origSender = sender()
       origSender ! workflowStatus(id)
-    case WorkflowOutputs(id) => workflowOutputs(id) pipeTo sender()
+    case NotifyCompletion(obj) => caller = Some(obj)
+    case WorkflowOutputs(id) => workflowOutputs(id) pipeTo sender() // FIXME: What if the workflow isn't done? How best to handle?
+    case Shutdown() => context.system.shutdown
     case WorkflowActor.Started => updateWorkflowState(sender(), WorkflowRunning)
-    case WorkflowActor.Done(symbolStore) => updateWorkflowState(sender(), WorkflowSucceeded)
-    case WorkflowActor.Failed(failures) => updateWorkflowState(sender(), WorkflowFailed)
+    case WorkflowActor.Done(symbolStore) =>
```

Comment:
Would be nice to have a func which contains these two lines and takes either `WorkflowSucceded` or `WorkflowFailed` and gets called by these two cases